src/main/webapp/user_profile_script.js

```diff
@@ -83,6 +89,8 @@ async function refresh() {
     loadMessages(document.getElementById("chat-id-input").value);
 }
 
+// To address the scaling issue and avoid loading too many messages at a time,
+// the chat box would first load at most 10 most recent messages.
```

Comment:
It seems like you are still fetching all the messages related to the task but only displayed the 10 most recent ones. To address the scaling issue, would it be better to only retrieve 10 messages from datastore at a time?

Reply:
Ah, that's a really good idea. I will use Cursor to solve this problem in the next commit.